src/boundaryCondition/ObstacleBoundary.cpp

```diff
@@ -219,35 +219,43 @@ namespace {
 void apply_boundary_condition(real *data, size_t **index_fields, const size_t *patch_starts,
                               const size_t *patch_ends, size_t level, BoundaryData *boundary_data,
                               size_t id, bool sync) {
-    for (size_t i = 0; i < number_of_patches; i++) {
-        size_t *d_patch = *(index_fields + i);
-        size_t patch_start = *(patch_starts + i);
-        size_t patch_end = *(patch_ends + i);
-        Patch p = static_cast<Patch>(i);
-        BoundaryCondition bc = boundary_data->get_boundary_condition(p);
-        switch (bc) {
-            case BoundaryCondition::DIRICHLET:
-                apply_dirichlet(data, d_patch, p, patch_start, patch_end, level,
-                                boundary_data->get_value(p));
-                break;
-            case BoundaryCondition::NEUMANN:
-                apply_neumann(data, d_patch, p, patch_start, patch_end, level,
-                              boundary_data->get_value(p));
-                break;
-            case BoundaryCondition::PERIODIC:
-                apply_periodic(data, d_patch, p, patch_start, patch_end, level, id);
-                break;
-            default:
 #ifndef BENCHMARKING
-                auto logger = Utility::create_logger("ObstacleBoundary");
-                logger->error("Unknown boundary condition: {}", bc);
+        auto logger = Utility::create_logger("ObstacleBoundary");
 #endif
-                break;
+        for (size_t i = 0; i < number_of_patches; i++) {
+            size_t *d_patch = *(index_fields + i);
+            size_t patch_start = *(patch_starts + i);
+            size_t patch_end = *(patch_ends + i);
+            auto p = static_cast<Patch>(i);
+            if (patch_start == patch_end) {
+#ifndef BENCHMARKING
+                logger->debug("skipping apply boundary condition for : {}", BoundaryData::get_patch_name(p));
+#endif
+                continue;
+            }
```

Comment:
this isn't a solid solution. we have to change the index lists in `multigrid.cpp` because at the moment there is no difference between a slice of 1 and slice of 0 as both have the same start and ending point.

Reply:
nvm, it does work, cause in `Multigrid.cpp:apply_boundary_condition` the first index of the next level is used for `patch_end`, resulting in the usage of ` < patch_end` which distinguished between a slice of 0 and a slice of 1.

slice_size of 0, no indices:
`patch_start`: 0
`patch_end`: 0
-> `for (size_t i = 0; i < 0; i++)` -> no loop iteration

slice_size of 1, index [0]:
`patch_start`: 0
`patch_end`: 1
-> `for (size_t i = 0; i < 0; i++)` -> 1 loop iteration with index 0